FreeRTOS/FreeRTOS-Plus-TCP
# IPv4/single: DHCP send option 50 only when a preferred IP-address is …

Description
-----------

A FreeRTOS forum subscriber [@aah](https://forums.freertos.org/u/aah) reported [here](https://forums.freertos.org/t/freertos-tcp-unexpected-new-dhcp-address-after-update/15364) that the DHCP option-50, should become optional during the discovery phase.

Option-50 indicates "the preferred or requested IP-address".
When it is supplied during "discovery", it is seen as a preferred address, in the "request" phase, it is seen as the requested IP-address.

The preferred address during "discovery" can be ignored by the DHCP server.

Test Steps
-----------
I tested the DHCP client in two situations:

1) Call `vDHCPSetPreferredIPAddress( FreeRTOS_inet_addr_quick( 192,168,1,10 ) )` before calling `FreeRTOS_IPInit()`
2) Just call `FreeRTOS_IPInit()` so that the preferred IP-address remains zero, i.e. not used.

Related Issue
-----------
When during discovery a preferred IP-address of zero is indicated, the DHCP server may have a different response, please refer to the post on the forum, mentioned here above.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
